site/content/docs/user/networking-scenarios.md

```diff
@@ -0,0 +1,228 @@
+
+We can solve it installing routes in the new containers to the Pod Subnets in each Node.
+
+### Example: Multiple network interfaces and Multi-Home Nodes
+
+There can be scenarios that requite multiple interfaces in the KIND nodes to test multi-homing, VLANS, CNI plugins, ... 
+
+Typically, you will want to use loopback addresses for communication. We can configure those loopback addresses after the cluster has been created, and then modify the Kubernetes components to use them.
+
+When creating the cluster we must add the loopback IP address of the control plane to the certificate SAN (the apiserver binds to "all-interfaces" by default):
+
+```yaml
+kind: Cluster
+apiVersion: kind.x-k8s.io/v1alpha4
+# add the loopback to apiServer cert SANS
+kubeadmConfigPatchesJSON6902:
+- group: kubeadm.k8s.io
+  kind: ClusterConfiguration
+  patch: |
+    - op: add
+      path: /apiServer/certSANs/-
+      value: my-loopback
+```
+
+In order to create the network interfaces, you can use tools like [koko](https://github.com/redhat-nfvpe/koko) to create new networking interfaces on the KIND nodes, you can check several examples of creating complex topologies with containers in this repo https://github.com/aojea/frr-lab.
+
+Other alternative is [using Docker user defined bridges](https://docs.docker.com/network/bridge/#connect-a-container-to-a-user-defined-bridge):
+
+```sh
+LOOPBACK_PREFIX="1.1.1."
+MY_BRIDGE="my_net2"
+MY_ROUTE=10.0.0.0/24
+MY_GW=172.16.17.1
+# Create 2nd network
+docker network create ${MY_BRIDGE}
+# Configure nodes to use the second network
+for n in $(kind get nodes); do
+  # Connect the node to the second network
+  docker network connect ${MY_BRIDGE} ${n}
+  # Configure a loopback address
+  docker exec ${n} ip addr add ${LOOPBACK_PREFIX}${i}/32 dev lo
+  # Add static routes
+  docker exec ${n} ip route add ${MY_ROUTE} via {$MY_GW}
+done
+```
+
+After the cluster has been created, we have to modify, in the control-plane node,  the kube-apiserver `--advertise-address` flag in the static pod manifest in `/etc/kubernetes/manifests/kube-apiserver.yaml` (once you write the file it restarts the pod with the new config):
+
+```yaml
+apiVersion: v1
+kind: Pod
+metadata:
+  creationTimestamp: null
+  labels:
+    component: kube-apiserver
+    tier: control-plane
+  name: kube-apiserver
+  namespace: kube-system
+spec:
+  containers:
+  - command:
+    - kube-apiserver
+    - --advertise-address=172.17.0.4
+```
+
+and then change in all the nodes the kubelet `node-ip` flag:
+
+```
+root@kind-worker:/# more /var/lib/kubelet/kubeadm-flags.env 
+KUBELET_KUBEADM_ARGS="--container-runtime=remote --container-runtime-endpoint=/run/containerd/containerd.sock --fail-swap-on=false --node-ip=172.17.0.4"
+ ```
+
+and restart them `systemctl restart kubelet` to use the new config
```

Comment:
Finally restart the kubelets to use the new configuration with `systemctl restart kubelet`.

Reply:
important to note here is that calling kubeadm init / join again on the node will override `/var/lib/kubelet/kubeadm-flags.env`. alternative is to use `/etc/default/kubelet`
https://kubernetes.io/docs/setup/production-environment/tools/kubeadm/kubelet-integration/#the-kubelet-drop-in-file-for-systemd

Reply:
let me add it as a note, due to the ephemeral nature of the nodes I don't expect people to issue those commands ... but :man_shrugging: